File: src/main/java/ru/ioleksiv/telegram/bot/core/model/method/sticker/DeleteStickerFromSet.java
package ru.ioleksiv.telegram.bot.core.model.method.sticker;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;
import ru.ioleksiv.telegram.bot.core.model.method.interfaces.RunnableAction;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;

/**
 * @see <a href="https://core.telegram.org/bots/api#deletestickerfromset>deleteStickerFromSet</a>
 */
public class DeleteStickerFromSet extends RunnableAction<Boolean> {
    private static final String METHOD = "deleteStickerFromSet";
    /**
     * sticker	String	Yes	File identifier of the sticker
     */
    @JsonProperty("sticker")
    private String sticker;

    public DeleteStickerFromSet(Networker networker) {
        super(METHOD, networker);
    }

    public String getSticker() {
        return sticker;
    }

    public DeleteStickerFromSet setSticker(String sticker) {
        this.sticker = sticker;
        return this;
    }

    @Override
   protected Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return ResponseCollection.BooleanResponse.class;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/sticker/DeleteStickerFromSet.kt
package io.github.oleksivio.telegram.bot.api.model.method.sticker

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.BooleanResponse
import io.github.oleksivio.telegram.bot.core.model.method.Action

/**
 * @see [deleteStickerFromSet](https://core.telegram.org/bots/api/#deletestickerfromset)
 */
class DeleteStickerFromSet(
        /**
         * sticker String Yes File identifier of the sticker
         */
        @JsonProperty(ApiDict.STICKER_KEY)
        val sticker: String
) : Action<Boolean>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "deleteStickerFromSet"

    @JsonIgnore
    override val resultWrapperClass = BooleanResponse::class
}


